mephisto/abstractions/providers/prolific/prolific_unit.py

```diff
@@ -0,0 +1,365 @@
+        return self.datastore.get_client_for_requester(requester_name)
+
+    @property
+    def log_prefix(self) -> str:
+        return f'[Unit {self.db_id}] '
+
+    def _sync_study_mapping(self) -> None:
+        """Sync with the datastore to see if any mappings have updated"""
+        if self.datastore.is_study_mapping_in_sync(self.db_id, self._last_sync_time):
+            return
+        try:
+            mapping = dict(self.datastore.get_study_mapping(self.db_id))
+            self.prolific_study_id = mapping['prolific_study_id']
+            self.prolific_submission_id = mapping.get('prolific_submission_id')
+            self.assignment_time_in_seconds = mapping.get('assignment_time_in_seconds')
+        except IndexError:
+            # Study does not appear to exist
+            self.prolific_study_id = None
+            self.prolific_submission_id = None
+            self.assignment_time_in_seconds = -1
+        # We update to a time slightly earlier than now, in order
+        # to reduce the risk of a race condition caching an old
+        # value the moment it's registered
+        self._last_sync_time = time.monotonic() - 1
+
+    def register_from_provider_data(
+        self, prolific_study_id: str, prolific_submission_id: str,
+    ) -> None:
+        """Update the datastore and local information from this registration"""
+        self.datastore.register_submission_to_study(
+            prolific_study_id=prolific_study_id,
+            unit_id=self.db_id,
+            prolific_submission_id=prolific_submission_id,
+        )
+        self._sync_study_mapping()
+
+    def get_prolific_submission_id(self) -> Optional[str]:
+        """Return the Prolific Submission ID (assignment ID) associated with this unit"""
+        self._sync_study_mapping()
+        return self.prolific_submission_id
+
+    def get_prolific_study_id(self) -> Optional[str]:
+        """Return the Porlific Study ID associated with this unit"""
+        self._sync_study_mapping()
+        return self.prolific_study_id
+
+    def get_requester(self) -> 'ProlificRequester':
+        """Wrapper around regular Requester as this will be ProlificRequester"""
+        if self.__requester is None:
+            self.__requester = cast('ProlificRequester', super().get_requester())
+        return self.__requester
+
+    def get_status(self) -> str:
+        """Get status for this unit directly from Prolific, fall back on local info"""
+
+        # -------------------------------------------------------------------------
+        # TODO (#1008): Fix this logic. It looked abstract and unclear in Mturk code.
+        if self.db_status == AssignmentState.CREATED:
+            return super().get_status()
+        elif self.db_status in [
+            AssignmentState.ACCEPTED,
+            AssignmentState.EXPIRED,
+            AssignmentState.SOFT_REJECTED,
+        ]:
+            # These statuses don't change with a get_status call
+            return self.db_status
```

Comment:
These are final status that won't be changing, once they're stored in the db that's it.